dhall-yaml/src/Dhall/Yaml.hs

```diff
@@ -54,25 +54,22 @@ jsonToYaml json documents quoted =
   case (documents, json) of
     (True, Data.Aeson.Array elems)
       -> Data.ByteString.intercalate "\n---\n"
-         $ fmap (Data.ByteString.Lazy.toStrict. (Data.YAML.Aeson.encodeValue' schemaEncoder YT.UTF8). (:[]))
-         $ Data.Vector.toList elems
-    _ -> Data.ByteString.Lazy.toStrict (Data.YAML.Aeson.encodeValue' schemaEncoder YT.UTF8 [json])
+         $ (Data.ByteString.Lazy.toStrict . encoder . (:[])) <$> Data.Vector.toList elems
+    _ -> Data.ByteString.Lazy.toStrict (encoder [json])
   where
     style (Y.SStr s)
         | "\n" `Text.isInfixOf` s =
             Right (YE.untagged, YE.Literal YE.Clip YE.IndentAuto, s)
         | quoted || Text.all isNumberOrDateRelated s || isBoolString =
             Right (YE.untagged, YE.SingleQuoted, s)
       where
-        isBoolString
-          | Text.length s > 5 = False
-          | otherwise =
-            case Text.toLower s of
-              "true" -> True
-              "false" -> True
-              _ -> False
+        -- For backwards compatibility with YAML 1.1, we need to add the following to the set of boolean values:
+        -- https://yaml.org/type/bool.html
+        isBoolString = Text.length s <= 5 &&
```

Comment:
I had been wondering whether taking the `length` of any string we check might itself be a performance problem – `Text.length` is O(n) after all.

Reply:
(Oops, I clicked "submit" too early.)

`text` has a rewrite rule that uses the `compareLength` function, which [IIUC](https://github.com/haskell/text/issues/284) should be O(5) in this case.

Reply:
Good to know.

I need to read/understand how Text is implemented. I had assumed a strict Text was just struct { arrayPtr, length } and that `length` was O(1)